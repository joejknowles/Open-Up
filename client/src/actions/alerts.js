export const removeAlert = (id) => ({
  type: 'REMOVE_ALERT',
  id
});
