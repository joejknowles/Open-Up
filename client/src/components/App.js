import React, { Component } from 'react';
import '../styles/App.css';
import DayBooking from './dayBooking';
import Errors from './alerts/errors';

import { Provider } from 'react-redux';

class App extends Component {
  render() {
    return (
      <Provider store={ this.props.store }>
        <div className="App">
          <Errors />
          <DayBooking />
        </div>
      </Provider>
    );
  }
}

export default App;
