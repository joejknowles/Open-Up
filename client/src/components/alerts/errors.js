import React from 'react';

export const Errors = ({ errors }) => (
  <div>
    {
      errors.map((error, index) => (
        <p key={ index }>{ error }</p>
      ))
    }
  </div>
);

export default Errors;
