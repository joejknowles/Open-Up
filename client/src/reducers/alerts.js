const isFailureType = ({ type }) => (
  type === 'FETCH_SLOTS_FAILURE' ||
  type === 'BOOK_SLOT_FAILURE'
);

export default (state = [], action) => {
  if (isFailureType(action)) {
    return [
      ...state,
      ...action.response.result
    ];
  }
  switch (action.type) {
    case 'REMOVE_ALERT':
      return [ ...state ].filter((id) => id !== action.id);
    case 'BOOK_SLOT_SUCCESS':
      return [ ...state, action.notificationId ];
    default:
      return state;
  }
};

export const alertsById = (state = {}, action) => {
  if (isFailureType(action)) {
    return {
      ...state,
      ...action.response.errors
    }
  }
  if (action.type === 'BOOK_SLOT_SUCCESS') {
    const newNotification = {
      id: action.notificationId,
      message: 'booked!',
      type: 'SUCCESS'
    };
    return {
      ...state,
      [action.notificationId]: newNotification
    }
  }
  return state;
};
