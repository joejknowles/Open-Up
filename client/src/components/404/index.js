import React from 'react';
import { Link } from 'react-router';
import '../../styles/App.css';

export default () => (
  <div className="App">
    <h3>Page not found</h3>
    <Link to="venue">Home</Link>
  </div>
);
