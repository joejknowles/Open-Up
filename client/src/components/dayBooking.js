import React, { Component } from 'react';
import '../styles/App.css';
import { connect } from 'react-redux';

export class DayBooking extends Component {
  render() {
    const date = this.props.date ?
      this.props.date.toLocaleDateString("en-GB") :
      '';
    const slots = this.props.slots || []
    return (
      <div>
        <h3>Today { date }</h3>
        { slots.map(({ id }) => (
            <button key={ id } ></button>
        )) }
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  slots: state.allSlots.map((id) => state.slotsById[id]),
  date: state.selectedDate
});

export default connect(mapStateToProps)(DayBooking);
