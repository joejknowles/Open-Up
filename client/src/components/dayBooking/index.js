import React, { Component } from 'react';
import '../../styles/App.css';
import { connect } from 'react-redux';
import { fetchSlots } from '../../actions';

import LoadingIndicator from '../loadingIndicator';
import Heading from './heading';
import List from './list';

export class DayBooking extends Component {
  componentDidMount() {
    this.props.fetchSlots();
  }

  render() {
    if (this.props.isLoading) return (<LoadingIndicator />);
    return (
      <div>
        <Heading />
        <List slots={ this.props.slots } />
      </div>
    );
  }
}

const mapStateToProps = (state) => ({
  slots: state.allSlots.map((id) => state.slotsById[id]),
  date: state.selectedDate,
  isLoading: state.isLoading
});

const mapDispatchToProps = (dispatch) => ({
  fetchSlots: fetchSlots(dispatch)
});

const mergeProps = ({ slots, isLoading, date }, dispatchProps) => ({
  slots, isLoading,
  fetchSlots: dispatchProps.fetchSlots(date)
});

export default connect(mapStateToProps, mapDispatchToProps, mergeProps)(DayBooking);
