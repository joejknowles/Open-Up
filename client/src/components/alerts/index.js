import React from 'react';
import { connect } from 'react-redux';
import ReactCSSTransitionsGroup from 'react-addons-css-transition-group';

import '../../styles/Alert.css';

import { alertsSelector, createAlertSelector } from '../../reducers';
import { removeAlert } from '../../actions/alerts'

export const Alert = ({ message, onClick, type }) => (
  <div
    className={ `Alert ${ type === 'ERROR' ? 'Error' : 'Success' }`}
    { ...{ onClick } }>
    { message }
  </div>
);

const mapDispatchToProps = (dispatch, { id }) => ({
  onClick: () => (dispatch(removeAlert(id)))
});
const mapStateToProps = (state, { id }) => createAlertSelector(id);

const ConnectedError = connect(
  mapStateToProps,
  mapDispatchToProps
)(Alert);

export const Alerts = ({ alerts }) => (
  <div className="Alerts">
    <ReactCSSTransitionsGroup
      transitionName="alert"
      transitionEnterTimeout={500}
      transitionLeaveTimeout={400}
      >
      {
        alerts.map((id) => (
          <ConnectedError key={ id } { ...{ id } } />
        ))
      }
    </ReactCSSTransitionsGroup>
  </div>
);

export default connect((state) => (
  { alerts: alertsSelector(state) }
))(Alerts);
