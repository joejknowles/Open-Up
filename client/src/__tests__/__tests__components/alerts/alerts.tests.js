import { Alerts, Alert } from '../../../components/alerts';

import React from 'react';
import ReactTestUtils from 'react-addons-test-utils';

const shallowRenderer = ReactTestUtils.createRenderer();

it('Alerts match snapshot', () => {
  const tree = shallowRenderer.render(
    <Alerts alerts={ ['1'] } />
  );
  expect(tree).toMatchSnapshot();
});

it('Alert matches snapshot with error', () => {
  const tree = shallowRenderer.render(
    <Alert message="error message" type="ERROR" />
  );
  expect(tree).toMatchSnapshot();
});

it('Alert matches snapshot with success notification', () => {
  const tree = shallowRenderer.render(
    <Alert message="error message" type="SUCCESS" />
  );
  expect(tree).toMatchSnapshot();
});
