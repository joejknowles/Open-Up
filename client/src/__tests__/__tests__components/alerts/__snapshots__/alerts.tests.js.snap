exports[`test Alert matches snapshot with error 1`] = `
<div
  className="Alert Error"
  onClick={undefined}>
  error message
</div>
`;

exports[`test Alert matches snapshot with success notification 1`] = `
<div
  className="Alert Success"
  onClick={undefined}>
  error message
</div>
`;

exports[`test Alerts match snapshot 1`] = `
<div
  className="Alerts">
  <ReactCSSTransitionGroup
    transitionAppear={false}
    transitionEnter={true}
    transitionEnterTimeout={500}
    transitionLeave={true}
    transitionLeaveTimeout={400}
    transitionName="alert">
    <Connect(Component)
      id="1" />
  </ReactCSSTransitionGroup>
</div>
`;
